orthoinference/src/main/java/org/reactome/orthoinference/PathwaysInferrer.java

```diff
@@ -0,0 +1,263 @@
+					}
+					infPathwayInst.addAttributeValue(inferredFrom, sourcePathwayReferralInst);
+					infPathwayInst.addAttributeValue(evidenceType, evidenceTypeInst);
+					for (GKInstance goBioProcessInst : (Collection<GKInstance>) sourcePathwayReferralInst.getAttributeValuesList(goBiologicalProcess))
+					{
+						infPathwayInst.addAttributeValue(goBiologicalProcess, goBioProcessInst);
+					}
+					infPathwayInst.addAttributeValue(orthologousEvent, sourcePathwayReferralInst);
+
+					if (sourcePathwayReferralInst.getSchemClass().isa(ReactionlikeEvent))
+					{
+						logger.warn(sourcePathwayReferralInst + " is a ReactionLikeEvent, which is unexpected -- refer to infer_events.pl");
+					}
+					infPathwayInst.setDisplayName(sourcePathwayReferralInst.getDisplayName());
+					inferredEventIdenticals.put(sourcePathwayReferralInst, infPathwayInst);
+					GKInstance orthoStableIdentifierInst = EventsInferrer.getStableIdentifierGenerator().generateOrthologousStableId(infPathwayInst, sourcePathwayReferralInst);
+					infPathwayInst.addAttributeValue(stableIdentifier, orthoStableIdentifierInst);
+					dba.storeInstance(infPathwayInst);
+
+					// This was replaced with addAttributeValueIfNecessary due to a bug where a Pathway instance's 'OrthologousEvent' attribute was being replaced,
+					// instead of being added to the existing array when  the script was executed from a jar (rather than from Eclipse) (Justin Cook 2018)
+					sourcePathwayReferralInst = InstanceUtilities.addAttributeValueIfNecessary(sourcePathwayReferralInst, infPathwayInst, orthologousEvent);
+					dba.updateInstanceAttribute(sourcePathwayReferralInst, orthologousEvent);
+
+					//TODO: At this point, sourcePathwayReferralInst is always a Pathway. Perhaps move to its own data structure? Holdout from Perl...
+					updatedInferrableHumanEvents.add(sourcePathwayReferralInst);
+				} else {
+					logger.info("Inferred Pathway instance already exists");
+				}
+				createInferredPathwayHierarchy(sourcePathwayReferralInst);
+			}
+		} else {
+			logger.info("Top Level Pathway inferred: " + sourceEventInst);
+		}
+	}
+
+	// This populates the hasEvent slot of all inferred Pathways that were just generated with corresponding inferred reactions
+	private static void addInferredEventsToInferredPathways() throws Exception {
+		Set<Long> seenInferredPathway = new HashSet<>();
+		for (GKInstance humanPathwayInst : updatedInferrableHumanEvents)
+		{
+			if (humanPathwayInst.getSchemClass().isValidAttribute(hasEvent)) {
+				if (!seenInferredPathway.contains(humanPathwayInst.getDBID())) {
+					// Collect inferred Events associated with source Event
+					List<GKInstance> inferredEventInstances = new ArrayList<>();
+					for (GKInstance eventInst : (Collection<GKInstance>) humanPathwayInst.getAttributeValuesList(hasEvent)) {
+						if (inferredEventIdenticals.get(eventInst) != null) {
+							inferredEventInstances.add(inferredEventIdenticals.get(eventInst));
+						}
+					}
```

Comment:
The for loop above could be its own method:
```
private List<GKInstance> getInferredEventInstances(GKInstance humanPathwayInst) {
    	// Collect inferred Events associated with source Event
	List<GKInstance> inferredEventInstances = new ArrayList<>();
	for (GKInstance eventInst : (Collection<GKInstance>) humanPathwayInst.getAttributeValuesList(hasEvent)) {
		if (inferredEventIdenticals.get(eventInst) != null) {
			inferredEventInstances.add(inferredEventIdenticals.get(eventInst));
		}
	}
}
```

Extracting related portions of a larger method into their own small methods helps to clarify what the function is doing overall by being more declarative than imperative.  It also helps find where the code that needs to be examined is by having a name so that section can be jumped to quickly from the main function rather than having to consider the entire function.